src/hypothesis/searchstrategy/types.py

```diff
@@ -0,0 +1,260 @@
+        known_type_strats.update({
+            np.dtype: nested_dtypes(),
+            np.ndarray: arrays(scalar_dtypes(), array_shapes(max_dims=2)),
+        })
+    except ImportError:
+        pass
+    return known_type_strats
+
+
+def type_sorting_key(t):
+    """Minimise to None, then non-container types, then container types."""
+    if t is None or t is type(None):
+        return -1
+    return issubclass(t, collections.abc.Container)
+
+
+def check_by_origin(thing, super_):
+    return thing is super_ or getattr(thing, '__origin__', None) is super_
+
+
+def try_issubclass(thing, maybe_superclass):
+    try:
+        return issubclass(thing, maybe_superclass)
+    except (AttributeError, TypeError):
+        # Some types can't be the subject or object of an instance or
+        # subclass check, including _TypeAlias and Union
+        return False
+
+
+def from_typing_type(thing):
+    import typing
+    # `Any` and `Type` mess up our subclass lookups, so handle them first
+    if thing is typing.Any:
+        raise ResolutionFailed('Cannot resolve typing.Any to any strategy.')
+    if check_by_origin(thing, getattr(typing, 'Type', object())):
+        if thing.__args__ in (None, typing.Any):
+            return st.just(type)
+        return st.from_type(thing.__args__[0]).map(type)
+    if check_by_origin(thing, typing.Union) or \
+            (sys.version_info[:2] == (3, 5) and thing.__name__ == 'Union'):
+        params = getattr(thing, '__union_params__', None) or ()
+        args = getattr(thing, '__args__', None) or ()
+        possible = sorted(params + args, key=type_sorting_key)
+        if not possible:
+            raise ResolutionFailed('Cannot resolve Union of no types.')
+        return st.one_of([st.from_type(t) for t in possible])
+    if isinstance(thing, typing.TypeVar):
+        if getattr(thing, '__contravariant__', False):
+            raise ResolutionFailed('Cannot resolve contravariant %s' % thing)
+        constraints = getattr(thing, '__constraints__', ())
+        if not constraints:
+            return st.builds(object)
+        # Pick a single constraint per run, and resolve it to a strategy
+        return st.shared(st.sampled_from(constraints), key=thing
+                         ).flatmap(st.from_type)
+    to_match = getattr(thing, '__origin__', None) or thing
+    # Of all types with a strategy, select the supertypes of this thing that
+    # whose subtypes have no strategy, and return their strategic union
+    mapping = dict(generic_type_strategy_mapping())
+    mapping.update({k: v for k, v in _global_type_to_strategy_lookup.items()
+                    if k.__module__ == 'typing'})
+    mapping = {k: v for k, v in mapping.items() if try_issubclass(k, to_match)}
+    if typing.Dict in mapping:
+        # View types are weird - the metaclasses are subclasses of (eg)
+        # Collection, but a concrete instance isn't an instance of Collection!
```

Comment:
Like almost everything, this of course varies with the version of the typing module!  From memory the tests fail under 3.5.1 (and maybe .2?) when I deleted this, and it would be very awkward to handle Collection taking one parameter, and resolving to a union including ItemsView which takes two.

(Also I might have meant Container.  Will investigate and fix or expand comment, at least.)